python/servo/command_base.py

```diff
@@ -671,52 +641,28 @@ def build_android_env_if_needed(self, env: Dict[str, str]):
         #
         # Also worth remembering: autoconf uses C for its configuration,
         # even for C++ builds, so the C flags need to line up with the C++ flags.
-        env['CFLAGS'] = ' '.join([
-            "--target=" + self.cross_compile_target,
-            "--sysroot=" + env['ANDROID_SYSROOT'],
-            "--gcc-toolchain=" + gcc_toolchain,
-            "-isystem", sysroot_include,
-            "-I" + arch_include,
-            "-B" + arch_libs,
-            "-L" + arch_libs,
-            "-D__ANDROID_API__=" + android_api,
-        ])
-        env['CXXFLAGS'] = ' '.join([
-            "--target=" + self.cross_compile_target,
-            "--sysroot=" + env['ANDROID_SYSROOT'],
-            "--gcc-toolchain=" + gcc_toolchain,
-            "-I" + cpufeatures_include,
-            "-I" + cxx_include,
-            "-I" + clang_include,
-            "-isystem", sysroot_include,
-            "-I" + cxxabi_include,
-            "-I" + clang_include,
-            "-I" + arch_include,
-            "-I" + support_include,
-            "-L" + gcc_libs,
-            "-B" + arch_libs,
-            "-L" + arch_libs,
-            "-D__ANDROID_API__=" + android_api,
-            "-D__STDC_CONSTANT_MACROS",
-            "-D__NDK_FPABI__=",
-        ])
-        env['CPPFLAGS'] = ' '.join([
-            "--target=" + self.cross_compile_target,
-            "--sysroot=" + env['ANDROID_SYSROOT'],
-            "-I" + arch_include,
-        ])
-        env["NDK_ANDROID_VERSION"] = android_api
+        env['CFLAGS'] = "--target=" + android_toolchain_name
+        env['CXXFLAGS'] = "--target=" + android_toolchain_name
+
+        # needed by mozjs
+        env['ANDROID_API_LEVEL'] = android_api
+        env["ANDROID_NDK_HOME"] = env["ANDROID_NDK_ROOT"]
```

Comment:
Hrm. mozjs should probably switch to using the same variable as Servo or vice versa? Which is more common? It looks like both are used?

Reply:
`ANDROID_NDK_ROOT` is  the official one now, so mozjs should be changed. Can that land separately?

Reply:
Sure. I can make the fix in mozjs.